### PR TITLE
pnpm minimumReleaseAgeExclude

### PR DESCRIPTION
**Pull request by ejfine**
## Why is this change necessary?
Need to be able to specify excluding patterns from the supply chain security limit

## How does this change address the issue?
Adds question and task to update pnpm-workspace.yaml in the grandchild template

## What side effects does this change have?
N/A

## How is this change tested?
Unit, and Downstream templates

## Other
Added CI check in this repo for answers file being present.  removed unneeded question about answers file in child template

## Summary by CodeRabbit

* **New Features**
  * Add configurable pnpm workspace exclusion support in the project template (prompt-driven) and a CLI task to apply patterns.

* **Tests**
  * Add unit tests and a test helper covering the new pnpm configuration behavior and edge cases.

* **Chores**
  * Add/adjust template tooling dependencies (including YAML handling).
  * CI: install Node during jobs and verify generated template answers exist.
  * Devcontainer/tooling: install pnpm and update devcontainer metadata and default pnpm/node versions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/LabAutomationAndScreening/copier-base-template/pull/162?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import argparse
```
Comment (zendern) [inline src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py:1]: So why all the parsing and such vs instead just calling `pnpm config --location project set minimumReleaseAgeExclude <you deps list comma separated>`?

Comment (ejfine) [reply src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py:1]: that would overwrite anything that someone had maunally added to pnpm-workspace.yaml though, right (we could mandate that "using the copier template is the only way to change minimumRelaseAgeExclude", but that potentially seems like a confusing user experience for someone at some point)?    also if you're instantiating a template the very first time in a blank devcontainer, pnpm might not be available (I don't think it is in the default Codespace devcontainer...but maybe now it is and that was just a while ago?)

Comment (ejfine) [reply src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py:1]: (obviously "I didn't think of that in the first place" is the real reason 😅    ...but this is me trying to evaluate both options and pros and cons right now)

Comment (zendern) [reply src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py:1]: I mean any time you pull in a copier update you have to deal with conflicts right? Wouldn't that just be one of those things you have to resolve at that time? 

> also if you're instantiating a template the very first time in a blank devcontainer, pnpm might not be available (I don't think it is in the default Codespace devcontainer...but maybe now it is and that was just a while ago?)

I mean i guess? That seems like an edge case for sure. And couldn't the task just blow up with "hey install pnpm" or at least have a big warning that states it didn't do a thing? 

But moreso hate that python doesn't have a default yaml parser lib cause this code is kinda gross due to it🙃

Comment (zendern) [reply src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py:1]: or another option if we decide we can live with the edge case of no pnpm installed that first time. 

`pnpm config --location project get minimum-release-age` call this to get the current value. Merge with your new value and call set.

**Comment by ejfine** [comment]
@coderabbitai full review

File: src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py (hunk 1 above)
Comment (ejfine) [reply src/copier_tasks/ensure_pnpm_minimum_release_age_exclude.py:1]: alright, took longer to get CI working (making sure pnpm was available) than it did to do the yaml manipulation in the first place 🤪 , but I think ultimately using the tool to update its own config is probably the better route

**Review by zendern — APPROVED** [review]
